Sources/Core/CodableReflection/ReflectionDecodable.swift

```diff
@@ -236,3 +236,26 @@ func forceCast<T>(_ type: T.Type) throws -> AnyReflectionDecodable.Type {
     }
     return casted
 }
+
+#if swift(>=4.2)
+extension ReflectionDecodable where Self: CaseIterable {
+
+    /// Default implementation of `ReflectionDecodable` for enums that are also `CaseIterable`.
+    ///
+    /// See `ReflectionDecodable.reflectDecoded(_:)` for more information.
+    public static func reflectDecoded() throws -> (Self, Self) {
+        /// enum must have at least 2 unique cases
+        guard !allCases.isEmpty, allCases.count > 1,
```

Comment:
Checking that the array is not empty _and_ checking the count is redundant.

Reply:
I know but isEmpty is more efficient though (enums are rarely big so I guess it doesn't really matter)